pkg/executor/docker/bidstrategy/semantic/image_platform.go

```diff
@@ -22,13 +22,13 @@ var _ bidstrategy.SemanticBidStrategy = (*ImagePlatformBidStrategy)(nil)
 var ManifestCache cache.Cache[docker.ImageManifest]
 var mu sync.Mutex
 
-func NewImagePlatformBidStrategy(client *docker.Client) *ImagePlatformBidStrategy {
+func NewImagePlatformBidStrategy(client *docker.Client, cfg config.Context) *ImagePlatformBidStrategy {
```

Comment:
we shouldn't be passing bacalhau config all the way down to these internal components, including executor, tracer, s3 storage provider and so on. The boundary for the config and `types.BacalhauConfig` should be the `node` package.

Reply:
I agree. Implementing that change as a part of this PR would have made this already large change even larger so I opted not to do it. I would prefer to do that in a follow on.